### PR TITLE
refactor(token): merges def and use of nft tokens

### LINKED ISSUE #659: Use config file for available tokens by Service
We are currently defining today in the `ui-config.json` file which are the Tokens accepted in each service. But the code is not retrieving those to populate the available currencies (at least in RNS). We would need to adapt the code to be the config file the source of truth of which tokens to populate/use in each service.

```
"services": {
            "rns": {
                "tokens": ["rif"]
            },
            "storage": {
                "tokens": ["rbtc", "rif"]
            }
        }
```

### PR DESCRIPTION
**Pull request by jurajpiar**
Removes obsolete definition and use of nft tokens and propagates latest token def and method across the project.

Resolves #659

### CODE REVIEW

File: src/components/pages/storage/buy/StorageOffersCheckoutPage.tsx [hunk 1]
```diff
@@ -85,7 +83,7 @@ const StorageOffersCheckoutPage: FC = () => {
 
   const {
     id,
-    token,
+    token: { symbol: token },
```
Comment (farinalvaro) [inline src/components/pages/storage/buy/StorageOffersCheckoutPage.tsx:86]:
```suggestion
	symbol: token
```
?

File: src/components/pages/rns/buy/DomainOffersCheckoutPage.tsx [hunk 2]
```diff
@@ -113,14 +113,14 @@ const DomainOffersCheckoutPage: FC<{}> = () => {
 
   // check funds
   useEffect(() => {
-    if (web3 && account && !isFundsConfirmed) {
+    if (web3 && account && !isFundsConfirmed && order) {
       const {
         item: {
-          paymentToken,
+          paymentToken: { symbol: tokenSymbol },
```
Comment (farinalvaro) [inline src/components/pages/rns/buy/DomainOffersCheckoutPage.tsx:119]:
```suggestion
          paymentToken: tokenSymbol,
```

File: src/api/rif-marketplace-cache/rns/common.ts [hunk 3]
```diff
@@ -24,17 +24,16 @@ export type RnsAPIService = Modify<
...
-export const isSupportedToken = (
+export const isSupportedRNSToken = (
   token: string,
-): boolean => SUPPORTED_TOKENS[token]
+): boolean => rnsSupportedTokens.some((t: string) => t === token)
 
-const { contractAddresses } = network
-contractAddresses.rbtc = ZERO_ADDRESS
+//  - Supported Token Address-to-token records
+export const rnsNftAddrTokenRecord: Record<string, BaseToken> = allNftAddresses
```
Comment (farinalvaro) [inline src/api/rif-marketplace-cache/rns/common.ts:32]:
```suggestion
export const rnsTokenAddressRecord: Record<string, BaseToken> = allNftAddresses
```

I think `allNftAddresses` should also be renamed in its file

**Review by itofarina — COMMENTED** [review]
Nice changes, just pointed some renaming that we discussed offline to avoid confusion.

File: src/components/pages/rns/buy/DomainOffersCheckoutPage.tsx (hunk 2 above)
Comment (jurajpiar) [reply src/components/pages/rns/buy/DomainOffersCheckoutPage.tsx:119]: paymentToken is type BaseToken

File: src/components/pages/storage/buy/StorageOffersCheckoutPage.tsx (hunk 1 above)
Comment (jurajpiar) [reply src/components/pages/storage/buy/StorageOffersCheckoutPage.tsx:86]: token is type BaseToken

File: src/contracts/storage/Storage.ts [hunk 4]
```diff
@@ -207,11 +214,11 @@ class StorageContract extends ContractWithTokens {
     {
       creatorOfAgreement,
       dataReferences = [],
-      token,
+      token: { symbol: token },
```
[the remark this answers is not in the document]
Comment (jurajpiar) [reply src/contracts/storage/Storage.ts:217]: token is type BaseToken

**Review by itofarina — APPROVED** [review]
All new code LGTM - didn't run it

**Pull request merged**